packages/fuels-types/src/bech32.rs

```diff
@@ -0,0 +1,102 @@
+use crate::errors::Error;
+use bech32::Variant::Bech32m;
+use bech32::{FromBase32, ToBase32};
+use core::str;
+use fuel_tx::Address;
+use std::fmt::{Display, Formatter};
+
+pub const FUEL_BECH32_HRP: &str = "fuel";
+
+#[derive(Debug, Clone)]
+pub struct Bech32Address {
+    plain_addr: Address,
+    encoding: String,
+}
+
+impl Bech32Address {
+    pub fn new(hrp: &str, hash: [u8; 32]) -> Self {
+        let data_base32 = hash.to_base32();
+        let encoding = bech32::encode(hrp, &data_base32, Bech32m).unwrap();
+
+        Self {
+            plain_addr: Address::from(hash),
+            encoding,
```

Comment:
I don't see the `encoding` field being used, can you please document what it represents.